js/main.js

```diff
@@ -0,0 +1,28 @@
+var $imageSubmit = document.querySelector('.image-submit');
+var $imageUpdate = document.querySelector('img');
+var $submitForm = document.querySelector('#user-entry-form');
+
+$imageSubmit.addEventListener('input', handlePhotoUpdate);
+$submitForm.addEventListener('submit', handleSubmit);
+
+function handlePhotoUpdate(event) {
+  $imageUpdate.setAttribute('src', event.target.value);
+}
+
+function handleSubmit(event) {
+  event.preventDefault();
+  var userEntryData = {};
+
+  userEntryData.title = $submitForm.elements.title.value;
+  userEntryData.url = $submitForm.elements.url.value;
+  userEntryData.notes = $submitForm.elements.notes.value;
+  userEntryData.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(userEntryData);
+
+  $imageUpdate.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $submitForm.reset();
+
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('user-entry-data-local-storage', dataJSON);
```

Comment:
You should be doing this for the `beforeunload` event, not every time the form is submitted. LocalStorage is a way to store data for future sessions, which means it only needs to be updated right before the current session ends. Also, please keep all localStorage-related code in `data.js`, in the same file as the variable it is preserving